docs/references/go/verifying-sessions.mdx

```diff
@@ -1,102 +1,161 @@
+
+The following example demonstrates how to use the `WithHeaderAuthorization()` middleware to protect a route. If the user tries accessing the route and their session token is valid, the user's ID will be returned in the response.
 
+<Callout type="info">
+Your Clerk secret key is required. If you are signed in to your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
+</Callout>
+
+<InjectKeys>
+```go
 import (
-	"net/http"
-	"strings"
+  "fmt"
+  "net/http"
 
-	"github.com/clerk/clerk-sdk-go/clerk"
+  "github.com/clerk/clerk-sdk-go/v2"
+  clerkhttp "github.com/clerk/clerk-sdk-go/v2/http"
 )
 
 func main() {
-	client, _ := clerk.NewClient(`{{secret}}`)
-
-	http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
-		// get session token from Authorization header
-		sessionToken := r.Header.Get("Authorization")
-		sessionToken = strings.TrimPrefix(sessionToken, "Bearer ")
-
-		// verify the session
-		sessClaims, err := client.VerifyToken(sessionToken)
-		if err != nil {
-			w.WriteHeader(http.StatusUnauthorized)
-			w.Write([]byte("Unauthorized"))
-			return
-		}
-
-		// get the user, and say welcome!
-		user, err := client.Users().Read(sessClaims.Claims.Subject)
-		if err != nil {
-			panic(err)
-		}
-
-		w.Write([]byte("Welcome " + *user.FirstName))
-	})
-
-	http.ListenAndServe(":8080", nil)
+  clerk.SetKey(`{{secret}}`)
+
+  mux := http.NewServeMux()
+  mux.HandleFunc("/", publicRoute)
+  protectedHandler := http.HandlerFunc(protectedRoute)
+  mux.Handle(
+    "/protected",
+    clerkhttp.WithHeaderAuthorization()(protectedHandler),
+  )
+
+  http.ListenAndServe(":3000", mux)
 }
-```
 
+func publicRoute(w http.ResponseWriter, r *http.Request) {
+  w.Write([]byte(`{"access": "public"}`))
+}
+
+func protectedRoute(w http.ResponseWriter, r *http.Request) {
+  claims, ok := clerk.SessionClaimsFromContext(r.Context())
+  if !ok {
+    w.WriteHeader(http.StatusUnauthorized)
+    w.Write([]byte(`{"access": "unauthorized"}`))
+    return
+  }
+  fmt.Fprintf(w, `{"user_id": "%s"}`, claims.Subject)
+}
+```
 </InjectKeys>
 
-## Using middleware
+## Manually verify a session token
 
-The Clerk SDK also provides a simple middleware that adds the active session to the request's context.
+Verifying a Clerk session manually is useful when you want to verify a session in a non-HTTP context, or if you would like total control over the verification process. With the solution above, Clerk middleware makes the minimum necessary requests to the Clerk Backend API. With this solution, you must be mindful of API rate limits.
```

Comment:
```suggestion
Verifying a Clerk session manually is useful when you want to verify a session in a non-HTTP context, or if you would like total control over the verification process. Clerk middleware makes the minimum necessary requests to the Clerk Backend API when you use it to verify sessions. When verifying manually, you must be mindful of API rate limits.
```

Suggested this change because this section may be moved around in the future without the text changing, at which point the "solution above" may no longer be above, which would confuse the reader